src/lib/common/utils.ts

```diff
@@ -0,0 +1,10 @@
+export function parseBool(payload?: string | null, defaultValue = false): boolean {
+  if (!payload) return defaultValue;
+
+  payload = payload.trim().toLowerCase();
+
+  // unlikely NaN will be passed, but included to be safe
+  const stringsOtherwiseTruthy = [0, false, NaN].map((v) => String(v).toLowerCase());
```

Comment:
![image](https://user-images.githubusercontent.com/8981287/169929561-3263a278-1582-4e60-9148-282864d33afc.png)

Update: I tested it

```ts
export function parseBool(payload?: string | null, defaultValue = false): boolean {
  if (!payload) return defaultValue;
  return !['0', 'false', 'nan'].includes(payload?.trim()?.toLowerCase());
}
```

Reply:
Since the `parseBool()` function is expected to run many times on the server, perhaps precomputing could increase the overall performance of the function. To that point, we probably shouldn't instantiate a new array `['0', 'false', 'nan']` each function call either. So to compromise, we can take advantage of precomputation while maintaining readability, I suggest we create a constant variable outside the `parseBool()` function body that runs once.
```ts
export const FALSE_TRUTHY_STRINGS = [0, false, NaN].map((v) => String(v).toLowerCase());

/** parseBool() */
```

How does this sound to you?

Reply:
I think we're talking nanoseconds of performance optimization at this point but. I believe that array would still get reinstantiated due to the stateless and ephemeral nature of serverless functions?

Reply:
This is true, but <https://github.com/EthanThatOneKid/acmcsuf.com/pull/455/commits/9e8decb4720a966d97dea67a611f34d0bc60d31b>.

Reply:
👍